File: src/algorithms/dynamic-programming/counting/pascals-triangle-row/sources/pascals_triangle_row_test.py
import importlib
import sys
import os

sys.path.insert(0, os.path.dirname(__file__))
mod = importlib.import_module("pascals-triangle-row")
pascals_triangle_row = mod.pascals_triangle_row

assert pascals_triangle_row(0) == [1], "row 0 should be [1]"
assert pascals_triangle_row(1) == [1, 1], "row 1 should be [1,1]"
assert pascals_triangle_row(2) == [1, 2, 1], "row 2 should be [1,2,1]"
assert pascals_triangle_row(3) == [1, 3, 3, 1], "row 3 should be [1,3,3,1]"
assert pascals_triangle_row(4) == [1, 4, 6, 4, 1], "row 4 should be [1,4,6,4,1]"
assert pascals_triangle_row(8) == [1, 8, 28, 56, 70, 56, 28, 8, 1], "row 8 check"

result6 = pascals_triangle_row(6)
assert len(result6) == 7, "row 6 should have length 7"

result5 = pascals_triangle_row(5)
assert result5[0] == 1 and result5[-1] == 1, "first and last should be 1"

result6_list = pascals_triangle_row(6)
row_sum = sum(result6_list)
assert row_sum == 64, "row 6 should sum to 64"

if __name__ == "__main__":
    print("All tests passed!")

File: src/algorithms/dynamic-programming/counting/pascals-triangle-row/sources/PascalsTriangleRow_test.java
// javac PascalsTriangleRow.java PascalsTriangleRow_test.java && java -ea PascalsTriangleRow_test
import java.util.Arrays;

public class PascalsTriangleRow_test {
    public static void main(String[] args) {
        assert Arrays.equals(PascalsTriangleRow.pascalsTriangleRow(0), new int[]{1}) : "row 0 should be [1]";
        assert Arrays.equals(PascalsTriangleRow.pascalsTriangleRow(1), new int[]{1, 1}) : "row 1 should be [1,1]";
        assert Arrays.equals(PascalsTriangleRow.pascalsTriangleRow(2), new int[]{1, 2, 1}) : "row 2 should be [1,2,1]";
        assert Arrays.equals(PascalsTriangleRow.pascalsTriangleRow(3), new int[]{1, 3, 3, 1}) : "row 3 check";
        assert Arrays.equals(PascalsTriangleRow.pascalsTriangleRow(4), new int[]{1, 4, 6, 4, 1}) : "row 4 check";
        assert Arrays.equals(PascalsTriangleRow.pascalsTriangleRow(8), new int[]{1, 8, 28, 56, 70, 56, 28, 8, 1}) : "row 8 check";

        int[] result6 = PascalsTriangleRow.pascalsTriangleRow(6);
        assert result6.length == 7 : "row 6 length should be 7";
        int rowSum = 0;
        for (int val : result6) rowSum += val;
        assert rowSum == 64 : "row 6 should sum to 64";

        System.out.println("All tests passed!");
    }
}
